On illumos CMAKE_SYSTEM_PROCESSOR for x64 is 'amd64' (see: `isainfo -k`).

Requires fixed CMake.

--- percona-server-8.0.13-3/storage/tokudb/CMakeLists.txt	2018-12-19 17:37:03.000000000 +0000
+++ percona-server-8.0.13-3/storage/tokudb/CMakeLists.txt.new	2019-01-13 08:48:57.355297297 +0000
@@ -37,8 +37,8 @@ ELSE ()
 ENDIF ()
 
 # check platform support, no 32 bit
-IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-  MESSAGE (${TOKUDB_STATUS_MODE} "x86_64 is only platform supported. ${CMAKE_SYSTEM_PROCESSOR} found. Not building TokuDB")
+IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64")
+  MESSAGE (${TOKUDB_STATUS_MODE} "amd64 is only platform supported. ${CMAKE_SYSTEM_PROCESSOR} found. Not building TokuDB")
   RETURN ()
 ENDIF ()
 
--- percona-server-8.0.13-3/storage/rocksdb/CMakeLists.txt	2018-12-19 17:31:11.000000000 +0000
+++ percona-server-8.0.13-3/storage/rocksdb/CMakeLists.txt.new	2019-01-13 08:48:47.790098703 +0000
@@ -22,8 +22,8 @@ ELSE ()
 ENDIF ()
 
 # check platform support, no 32 bit
-IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-  MESSAGE (${MYROCKS_STATUS_MODE} "x86_64 is only platform supported. ${CMAKE_SYSTEM_PROCESSOR} found. Not building MyRocks")
+IF (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64")
+  MESSAGE (${MYROCKS_STATUS_MODE} "amd64 is only platform supported. ${CMAKE_SYSTEM_PROCESSOR} found. Not building MyRocks")
   RETURN ()
 ENDIF ()
 
